sdk/state/ingest_test.go

```diff
@@ -64,16 +64,142 @@ func TestChannel_IngestTx_updateUnauthorizedCloseAgreement(t *testing.T) {
+	require.NoError(t, err)
+	_, err = initiatorChannel.ConfirmOpen(open)
+	require.NoError(t, err)
+
+	// Pretend responder broadcasts latest declTx, and
+	// initiator ingests it.
+	declTx, _, err := responderChannel.CloseTxs()
+	require.NoError(t, err)
+	err = initiatorChannel.IngestTx(declTx)
+	require.NoError(t, err)
+	closeState, err := initiatorChannel.CloseState()
+	require.NoError(t, err)
+	require.Equal(t, CloseStateClosing, closeState)
+
+	// TODO - initiator should not be able to propose/confirm new close agreements
+}
+
+func TestChannel_IngestTx_oldDeclTx(t *testing.T) {
+	// Setup
+	initiatorSigner := keypair.MustRandom()
+	responderSigner := keypair.MustRandom()
+	initiatorEscrowAccount := &EscrowAccount{
+		Address:        keypair.MustRandom().FromAddress(),
+		SequenceNumber: int64(101),
+	}
+	responderEscrowAccount := &EscrowAccount{
+		Address:        keypair.MustRandom().FromAddress(),
+		SequenceNumber: int64(202),
+	}
+	initiatorChannel := NewChannel(Config{
+		NetworkPassphrase:   network.TestNetworkPassphrase,
+		MaxOpenExpiry:       time.Hour,
+		Initiator:           true,
+		LocalSigner:         initiatorSigner,
+		RemoteSigner:        responderSigner.FromAddress(),
+		LocalEscrowAccount:  initiatorEscrowAccount,
+		RemoteEscrowAccount: responderEscrowAccount,
+	})
+	responderChannel := NewChannel(Config{
+		NetworkPassphrase:   network.TestNetworkPassphrase,
+		MaxOpenExpiry:       time.Hour,
+		Initiator:           false,
+		LocalSigner:         responderSigner,
+		RemoteSigner:        initiatorSigner.FromAddress(),
+		LocalEscrowAccount:  responderEscrowAccount,
+		RemoteEscrowAccount: initiatorEscrowAccount,
+	})
+	open, err := initiatorChannel.ProposeOpen(OpenParams{
+		ObservationPeriodTime:      1,
+		ObservationPeriodLedgerGap: 1,
+		ExpiresAt:                  time.Now().Add(time.Minute),
+	})
+	require.NoError(t, err)
+	open, err = responderChannel.ConfirmOpen(open)
+	require.NoError(t, err)
+	_, err = initiatorChannel.ConfirmOpen(open)
+	require.NoError(t, err)
+	initiatorChannel.UpdateLocalEscrowAccountBalance(100)
+	initiatorChannel.UpdateRemoteEscrowAccountBalance(100)
+	responderChannel.UpdateLocalEscrowAccountBalance(100)
+	responderChannel.UpdateRemoteEscrowAccountBalance(100)
+
+	oldDeclTx, _, err := responderChannel.CloseTxs()
+	require.NoError(t, err)
+
+	// New payment.
+	close, err := initiatorChannel.ProposePayment(8)
+	require.NoError(t, err)
+	close, err = responderChannel.ConfirmPayment(close)
+	require.NoError(t, err)
+	_, err = initiatorChannel.ConfirmPayment(close)
+	require.NoError(t, err)
+
+	// Pretend that responder broadcasts the old declTx, and
+	// initiator ingests it.
+	err = initiatorChannel.IngestTx(oldDeclTx)
+	require.NoError(t, err)
+	closeState, err := initiatorChannel.CloseState()
+	require.NoError(t, err)
+	require.Equal(t, CloseStateNeedsClosing, closeState)
+
+	// TODO - initiator should not be able to propose/confirm new close agreements
```

Comment:
FYI I don't think you need to test this in this test, it's sufficient to test that the state is set appropriately, but if you want to I think it's fine too.

Reply:
ya  I guess that is covered in the other unit test coverage https://github.com/stellar/experimental-payment-channels/pull/232/commits/93bdc78262142a14ffd5d11c9c1d65a98fe73b70

it was more of a reminder to make sure those validations check against this new state, but that's noted in this [ticket](https://github.com/stellar/experimental-payment-channels/issues/233#issue-967612416)